test/analysis/test_ml.py

```diff
@@ -0,0 +1,28 @@
+import filecmp
+import pandas as pd
+import src.analysis.ml.ml as ml
+
+
+TEST_DIR = 'test/analysis/input/test/'
+OUT_DIR = 'test/analysis/output'
+EXPECT_DIR = 'test/analysis/expected'
+
+class TestML:
+
+    def test_summarize_networks(self):
+        dataframe = ml.summarize_networks([TEST_DIR+'s1.txt',TEST_DIR+'s2.txt',TEST_DIR+'s3.txt'])
+        dataframe.to_csv(OUT_DIR+'/dataframe.csv')
+        assert (filecmp.cmp(OUT_DIR+'/dataframe.csv', EXPECT_DIR+'/expected_df.csv'))
+
+    def test_pca(self):
+        dataframe = ml.summarize_networks([TEST_DIR+'s1.txt',TEST_DIR+'s2.txt',TEST_DIR+'s3.txt'])
+        ml.pca(dataframe, OUT_DIR+'/pca.png', OUT_DIR+'/pca-components.txt', OUT_DIR+ '/pca-coordinates.txt')
+        assert(filecmp.cmp(OUT_DIR+ '/pca-coordinates.txt', EXPECT_DIR+ '/expected_coords.txt'))
```

Comment:
The test case may be failing due to slight floating point differences. We can instead:
- Load pca-coordinates.txt as a dataframe
- Load expected_coords.txt as a dataframe
- Round both to 5 decimal places
- Check they match with `assert df1.equals(df2)`